### PR TITLE
get_cable_node() twix

### PR DESCRIPTION
**Pull request by Dennok**
get_cable_node can take custom layer to check.
cable layers bitwise suitable

Some more flexibility

## Changelog
:cl:
code: get_cable_node() can take custom layer to check
code: cable layers bitwise suitable
/:cl:

### CODE REVIEW

File: code/modules/power/power.dm [hunk 1]
```diff
@@ -354,12 +354,12 @@
...
 
 // return a cable if there's one on the turf, null if there isn't one
-/turf/proc/get_cable_node()
+/turf/proc/get_cable_node(check_layer = CABLE_LAYER_2)
 	if(!can_have_cabling())
 		return null
 	var/obj/structure/cable_bridge/B = locate() in src
 	for(var/obj/structure/cable/C in src)
-		if(C.cable_layer == CABLE_LAYER_2 || B)
+		if(B || C.cable_layer & check_layer)
```
Comment (Dennok) [inline code/modules/power/power.dm]:
```suggestion
		if(C.cable_layer & check_layer || B)
```

File: code/modules/power/power.dm [hunk 2]
```diff
@@ -354,12 +354,12 @@
...
 
 // return a cable if there's one on the turf, null if there isn't one
-/turf/proc/get_cable_node()
+/turf/proc/get_cable_node(check_layer = CABLE_LAYER_2)
 	if(!can_have_cabling())
 		return null
 	var/obj/structure/cable_bridge/B = locate() in src
 	for(var/obj/structure/cable/C in src)
-		if(C.cable_layer == CABLE_LAYER_2 || B)
+		if(C.cable_layer & check_layer || B)
```
Comment (spookydonut) [inline code/modules/power/power.dm]:
```
#define CABLE_LAYER_1 "l1"
#define CABLE_LAYER_2 "l2"
#define CABLE_LAYER_3 "l3"
```
you cant do bitwise on these

Comment (Dennok) [reply code/modules/power/power.dm]: Now cable layers bitwise suitable

File: code/__DEFINES/power.dm [hunk 3]
```diff
@@ -1,7 +1,7 @@
-#define CABLE_LAYER_1 "l1"
-#define CABLE_LAYER_2 "l2"
-#define CABLE_LAYER_3 "l3"
+#define CABLE_LAYER_1		0
+#define CABLE_LAYER_2		1
+#define CABLE_LAYER_3		2
```
Comment (Dennok) [inline code/__DEFINES/power.dm]:
```suggestion
#define CABLE_LAYER_1		1
#define CABLE_LAYER_2		2
#define CABLE_LAYER_3		4
```

**Review by kriskog — APPROVED** [review]

**Pull request merged**